src/phoenix/experimental/evals/functions/classify.py

```diff
@@ -404,48 +402,42 @@ def run_evals(
 
     async def _arun_eval(
         payload: RunEvalsPayload,
-    ) -> Tuple[EvaluatorIndex, RowIndex, Label, Score, Explanation]:
-        label, score, explanation = await payload.evaluator.aevaluate(
+    ) -> Tuple[Label, Score, Explanation]:
+        return await payload.evaluator.aevaluate(
             payload.record,
             provide_explanation=provide_explanation,
             use_function_calling_if_available=use_function_calling_if_available,
         )
-        return payload.evaluator_index, payload.row_index, label, score, explanation
 
     def _run_eval(
         payload: RunEvalsPayload,
-    ) -> Tuple[EvaluatorIndex, RowIndex, Label, Score, Explanation]:
-        label, score, explanation = payload.evaluator.evaluate(
+    ) -> Tuple[Label, Score, Explanation]:
+        return payload.evaluator.evaluate(
             payload.record,
             provide_explanation=provide_explanation,
             use_function_calling_if_available=use_function_calling_if_available,
         )
-        return payload.evaluator_index, payload.row_index, label, score, explanation
 
     executor = get_executor_on_sync_context(
         _run_eval,
         _arun_eval,
         concurrency=concurrency,
         tqdm_bar_format=get_tqdm_progress_bar_formatter("run_evals"),
         exit_on_error=True,
-        fallback_return_value=(None, None),
+        fallback_return_value=(None, None, None),
     )
+
+    total_records = len(dataframe)
     payloads = [
-        RunEvalsPayload(
-            evaluator_index=evaluator_index,
-            row_index=row_index,
-            evaluator=evaluator,
-            record=row.to_dict(),
-        )
-        # use the position of the row rather than the dataframe index, which is used
-        # to ensure the output dataframe has the same row order as the input dataframe
-        for row_index, (_, row) in enumerate(dataframe.iterrows())
-        for evaluator_index, evaluator in enumerate(evaluators)
+        RunEvalsPayload(evaluator=evaluator, record=row)
+        for evaluator, (_, row) in product(evaluators, dataframe.iterrows())
     ]
     eval_results: List[DefaultDict[RowIndex, Dict[ColumnName, Union[Label, Explanation]]]] = [
```

Comment:
`RowIndex` is confusing in this context, since this index no longer refers to the row of the dataframe. `int` is fine, as is a `Index` or `PayloadIndex` type alias.

Reply:
How is the index different now? I believe the structure of eval_results is identical to how it was previously (hence why none of the types have changed)